api/RS.API/RS.Service/Logic/CandidateManagerService.cs

```diff
@@ -4,19 +4,171 @@
+            catch (Exception e)
+            {
+                result.Message = e.Message;
+                result.Status = Status.Error;
+            }
+            return result;
+        }
+
+        public IResult GetCandidateById(Guid id)
+        {
+            var result = new Result
+            {
+                Operation = Operation.Read,
+                Status = Status.Success
+            };
+            try
+            {
+                var candidate = new CandidateViewModel();
+                var getCandidate = _candidateRepository.GetByID(id);
+                var openingCandidate = _candidateRepository.GetOpeningCandidate(getCandidate.CandidateId);
+                candidate.Opening = openingCandidate.Opening.OpeningId;
+                candidate.Qualification = getCandidate.Qualification.QualificationId;
+                candidate.Organization = getCandidate.Organisation.Name;
+                result.Body = candidate.MapFromModel(getCandidate);
+            }
+            catch (Exception e)
+            {
+                result.Message = e.Message;
+                result.Status = Status.Error;
+            }
+            return result;
+        }
+
+        public IResult UpdateCandidate(CandidateViewModel candidate)
+        {
+            var result = new Result
+            {
+                Operation = Operation.Update,
+                Status = Status.Success
+            };
+            try
+            {
+                var candidateModel = new Candidates();
+                candidateModel.MapFromViewModel(candidate, (ClaimsIdentity)_principal.Identity);
+                candidateModel.QualificationId = candidate.Qualification;
+                candidateModel.Qualification = _qualificationRepository.GetByID(candidate.Qualification);
+
+                var openingCandidate = _candidateRepository.GetOpeningCandidate(candidate.CandidateId);
+                var updatedOpeningCandidate = new OpeningCandidates();
+                if ((openingCandidate != null) && (openingCandidate.Opening.OpeningId != candidate.Opening))
+                {
+                    openingCandidate.MapDeleteColumns((ClaimsIdentity)_principal.Identity);
+                    updatedOpeningCandidate.OpeningId = candidate.Opening;
+                    updatedOpeningCandidate.candidate = candidateModel;
+                    updatedOpeningCandidate.Opening = _openingRepository.GetByID(candidate.Opening);
+                    updatedOpeningCandidate.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+                }
+                var organization = new Organizations();
+                organization.Name = candidate.Organization;
+                organization.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+
+                _candidateRepository.UpdateCandidate(candidateModel, updatedOpeningCandidate, organization);
+                _candidateRepository.Update(candidateModel);
+                _candidateRepository.SaveChanges();
```

Comment:
made the same changes as suggested for add candidate

Reply:
Done